### ISSUE TITLE
`useAccordionItem` and `useSelectableItem` have conflicting `usePress` props

### ISSUE THREAD

**Issue by justinsunho** [adobe/react-spectrum#4984]
### Provide a general summary of the issue here

When using `useAccordionItem` I noticed that I get the `console.warn` message `onClick is deprecated, please use onPress` when I click my accordion item, and this message `stopPropagation is now the default behavior for events in React Spectrum. You can use continuePropagation() to revert this behavior.` when I use keyboard interactions. I traced this down to how `useAccordionItem` uses `useSelectableItem`.

The `onPress` event in `useAccordionItem` works. However, an `onClick` event in the `itemProps` returned by `useSelectableItem` is being merged with said `onPress` prop, making it so that keyboard press events aren't usable and logs are filled with the above console warn.

I've traced all of this from a `usePress` being used in `useSelectableItem`. This seems to be working in conflict with the `useButton` and `onPress` in `useAccordionItem`

### 🤔 Expected Behavior?

* There should be no console warnings for onClick events that are used.
* Keyboard events should be usable when using `useAccordionITem`

### 😯 Current Behavior

One can see the console errors here: https://q9m1p6.csb.app/ Although this codesandbox seems to be an example that was used to play around with, it still uses the same `useSelectableItem` in its `useAccordion` and it has the same warnings + wrong keyboard events.

### 💁 Possible Solution

I don't know the codebase well enough to suggest a simple solution. However, I would like it if the `onPress` used in `useAccordionItem` would just override the `pressProps` values in `useSelectableItem` or if there would be clarity between how `useAccordionItem` and `useSelectableItem` are used.

### 🔦 Context

I am trying to build a reusable `Accordion` component using `useAccordion` and `useAccordionItem`

### 🖥️ Steps to Reproduce

https://q9m1p6.csb.app/ 

Simply implement an Accordion using `useAccordion` and `useAccordionItem`.

```
function Accordion(props) {
  const domRef = useRef<HTMLDivElement>();

  const state = useTreeState({ ...props, selectionMode: 'single' });
  const { accordionProps } = useAccordion(props, state, domRef);

  return (
    <div ref={domRef} {...accordionProps}>
      {Array.from(state.collection).map((item) => (
        <AccordionItem key={item.key} item={item} state={state} />
      ))}
    </div>
  );
}

function AccordionItem(prop) {
  const domRef = useRef<HTMLButtonElement>();
  const { state, item } = props;
  const { buttonProps, regionProps } = useAccordionItem(props, state, domRef);

  const isOpen = state.expandedKeys.has(item.key);

  return (
    <div>
      <h4 className="flex items-center w-full">
        <button
          {...buttonProps}
          ref={domRef}
          className="flex items-center justify-between w-full"
        >
          {item.props.title}
        </button>
      </h4>
      <div {...regionProps}>
        {item.props.children} {isOpen ? 'open' : 'closed'}
      </div>
    </div>
  );
}

<Accordion selectionMode="single">
  <Item key="files" title="Your files">
    files
  </Item>
  <Item key="shared" title="Shared with you">
    shared
  </Item>
  <Item key="last" title="Last item">
    last
  </Item>
</Accordion>
```

### Version

@react-aria/accordion: 3.0.0-alpha.20, react-aria: 3.22.0, react-stately: 3.20.0

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

MacOS

**Comment by LFDanLu** [comment]
I don't think you need `selectionMode` defined for your Accordion nor in the `useTreeState`. The tree state can track its expanded keys without a `selectionMode` being defined and selection tracking doesn't really make sense for a Accordion in the first place since you aren't selecting the rows persay.
https://codesandbox.io/s/peaceful-mclaren-wk6778?file=/src/App.js

**Comment by justinsunho** [comment]
@LFDanLu actually, I _just_ tested with if it was the `selectionMode`, getting rid of it solves the issue. My only question would be, how would we be able to make it so that an `AccordionItem` collapses if another `AccordionItem` is selected? Should we use a custom prop?

**Comment by LFDanLu** [comment]
At the moment `useTreeState` doesn't support enforcing that only a single key can be expanded unfortunately. I think in this case we would either add support for that in `useTreeState` (aka something like `expansionMode: 'single' | 'multiple'?` or create `useAccordionState` that calls `useTreeState` but overrides its `state.toggleKey` in such a way that it enforces that only a single expanded key can be set:
```
// psudocode
function toggleKey(set: Set<Key>, key: Key): Set<Key> {
  let res = new Set(set);
  if (res.has(key)) {
    res.delete(key);
  } else {
    res.clear();
    res.add(key)
  }

  return res;
}
```

**Comment by justinsunho** [comment]
that sounds like a great idea! Thanks for mentioning that `useTreeState` doesn't currently support this feature, gives a lot of clarity for where to go from here. 

I'll explore those options on my own and will hopefully find the time to open a PR. All my questions/ problems were solved so I'll close for now.
